### PR TITLE
162 eip712 domain separator info should be set by user

### PR DESCRIPTION
**Pull request by ColePBryan**

### CODE REVIEW

File: tap_aggregator/src/server.rs [hunk 1]
```diff
@@ -182,7 +185,18 @@ impl RpcServer for RpcImpl {
         let receipts_grt: u128 = receipts.iter().map(|r| r.message.value).sum();
         let receipts_count: u64 = receipts.len() as u64;
 
-        match aggregate_receipts_(api_version, &self.wallet, receipts, previous_rav).await {
+        //todo: remove clone from domain separator below. This should be complete before this makes it to code review! If this is still here then I forgot...
+        //      if this makes it into the actual code then the reviewers should be ashamed of themselves for not catching this.(This last part was a suggested comment from Copilot)
+
```
Comment (aasseman) [inline tap_aggregator/src/server.rs:190]: You didn't forget! But you forgot to remove the comment itself :p

File: tap_aggregator/src/main.rs [hunk 2]
```diff
@@ -95,3 +124,21 @@ async fn main() -> Result<()> {
...
+fn create_eip712_domain(args: &Args) -> Eip712Domain {
+    // Transfrom the args into the types expected by Eip712Domain::new().
+
+    // Transform optional strings into optional Cow<str>.
+    let name = args.domain_name.clone().map(Cow::Owned);
+    let version = args.domain_version.clone().map(Cow::Owned);
+
+    // Transform optional strings into optional U256.
+    let chain_id: Option<U256> = args.domain_chain_id.as_ref().map(|s| s.parse().unwrap());
+    let salt: Option<FixedBytes<32>> = args.domain_salt.as_ref().map(|s| s.parse().unwrap());
```
Comment (aasseman) [inline tap_aggregator/src/main.rs:137]: To help the user undestand what went wrong.
```suggestion
    let chain_id: Option<U256> = args.domain_chain_id.as_ref().map(|s| s.parse().expect("Domain chain ID"));
    let salt: Option<FixedBytes<32>> = args.domain_salt.as_ref().map(|s| s.parse().expect("Domain salt"));
```

**Review by aasseman — CHANGES_REQUESTED** [review]

File: tap_aggregator/src/server.rs (hunk 1 above)
Comment (ColePBryan) [reply tap_aggregator/src/server.rs:190]: Well at least you caught it 😄

File: tap_aggregator/src/main.rs (hunk 2 above)
Comment (ColePBryan) [reply tap_aggregator/src/main.rs]: I fixed this by propagating the error instead. Let me know if you would prefer an `expect` there.

File: tap_aggregator/src/main.rs [hunk 3]
```diff
@@ -42,6 +47,26 @@ struct Args {
...
+    #[arg(long, env = "TAP_DOMAIN_VERSION")]
+    domain_version: Option<String>,
+
+    /// Domain chain ID to be used for the EIP-712 domain separator.
+    #[arg(long, env = "TAP_DOMAIN_CHAIN_ID")]
+    domain_chain_id: Option<String>,
+
+    /// Domain verifying contract to be used for the EIP-712 domain separator.
+    #[arg(long, env = "TAP_DOMAIN_VERIFYING_CONTRACT")]
+    domain_verifying_contract: Option<Address>,
```
Comment (aasseman) [inline tap_aggregator/src/main.rs:65]: Shouldn't these be mandatory arguments?

File: tap_aggregator/src/main.rs (hunk 2 above)
Comment (aasseman) [reply tap_aggregator/src/main.rs]: Even better!

File: tap_aggregator/src/main.rs (hunk 3 above)
Comment (ColePBryan) [reply tap_aggregator/src/main.rs:65]: Technically they are not required by eip712. Any or all of them could be skipped.

Comment (aasseman) [reply tap_aggregator/src/main.rs:65]: Noted, thanks!

**Review by aasseman — APPROVED** [review]

**Pull request merged**